velox/connectors/hive/SplitReader.cpp

```diff
@@ -276,20 +276,44 @@ void SplitReader::createRowReader(
     std::shared_ptr<common::MetadataFilter> metadataFilter) {
   auto& fileType = baseReader_->rowType();
   auto columnTypes = adaptColumns(fileType, baseReaderOpts_.getFileSchema());
+  auto columnNames = fileType->names();
 
+  setRowIndexColumnInfoIfNeed(fileType, columnNames, columnTypes);
   configureRowReaderOptions(
       baseRowReaderOpts_,
       hiveTableHandle_->tableParameters(),
       scanSpec_,
       metadataFilter,
-      ROW(std::vector<std::string>(fileType->names()), std::move(columnTypes)),
+      ROW(std::vector<std::string>(columnNames), std::move(columnTypes)),
       hiveSplit_);
   // NOTE: we firstly reset the finished 'baseRowReader_' of previous split
   // before setting up for the next one to avoid doubling the peak memory usage.
   baseRowReader_.reset();
   baseRowReader_ = baseReader_->createRowReader(baseRowReaderOpts_);
 }
 
+void SplitReader::setRowIndexColumnInfoIfNeed(
+    const RowTypePtr& fileType,
+    std::vector<std::string>& columnNames,
+    std::vector<facebook::velox::TypePtr>& columnTypes) {
+  auto rowIndexMetaColIdx =
+      readerOutputType_->getChildIdxIfExists(kSparkReservedTmpMetaRowIndex);
+  if (rowIndexMetaColIdx.has_value() &&
```

Comment:
We should also add an option in `HiveConfig` to enable this behavior, just in case if we actually have this column in file in other engines.